WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift

```diff
@@ -386,8 +386,10 @@ extension SiteStatsInsightsTableViewController: SiteStatsInsightsDelegate {
             return
         }
 
-        let postingActivityViewController = PostingActivityViewController.loadFromStoryboard()
-        postingActivityViewController.yearData = viewModel.yearlyPostingActivity()
+        let postingActivityViewModel = PostingActivityViewModel(insightsStore: insightsStore)
+        let postingActivityViewController = PostingActivityViewController.loadFromStoryboard { coder in
```

Comment:
This change doesn't seem to affect when the expensive calculation is performed. Instead of line 390, it gets called on `pushViewController`. But with the other performance improvements made in this PR, it should be acceptably fat regardless of how many posts you have.